util/datasize/datasize.go

```diff
@@ -0,0 +1,331 @@
+	if b == 0 {
+		return fmt.Sprint("0B")
+	}
+	for _, dimension := range dimensions {
+		if dimension.flags&flags == flags && b > dimension.size {
+			return fmt.Sprintf("%.3f %s", b.dimensionCount(dimension.size), dimension.spelling)
+		}
+	}
+	return fmt.Sprintf("%d bit", b)
+}
+
+func (b bitSize) PreciseString(flags uint8) string {
+	if b == 0 {
+		return fmt.Sprint("0B")
+	}
+	for _, dimension := range dimensions {
+		if dimension.flags&flags == flags && b%dimension.size == 0 {
+			return fmt.Sprintf("%d %s", b/dimension.size, dimension.spelling)
+		}
+	}
+	return fmt.Sprintf("%d bit", b)
+}
+
+func splitDimension(text []byte) (dimension string, size interface{}, err error) {
+	str := string(text)
+
+	parts := strings.FieldsFunc(str, func(c rune) bool {
+		return c == ' '
+	})
+
+	var sizeStr string
+
+	//TODO: Is it ok? empty string is being parsed as default(0) value.
+	if len(parts) == 0 {
+		return "", uint64(0), nil
+	}
+	if len(parts) == 1 {
+		dimension = strings.TrimLeftFunc(parts[0], func(r rune) bool {
+			return '0' <= r && r <= '9'
+		})
+		sizeStr = strings.TrimRightFunc(parts[0], func(r rune) bool {
+			return '0' > r || r > '9'
+		})
+	} else {
+		if len(parts) != 2 {
+			err = fmt.Errorf(`could not parse bitSize - "%s" can not be split to 2 parts`, str)
+			return
+		}
+		dimension = parts[1]
+		sizeStr = parts[0]
+	}
+
+	size, err = strconv.ParseUint(sizeStr, 10, 64)
+	if err != nil {
+		size, err = strconv.ParseFloat(parts[0], 64)
+		if err != nil {
+			err = fmt.Errorf("could not parse bitSize numeric part -  %s", err)
+		}
+	}
+	return
+}
+
+func getPossibleDimensions(flags uint8) []string {
+	result := []string{}
+	for _, dim := range dimensions {
+		if dim.flags&flags == flags {
+			result = append(result, dim.spelling)
+		}
+	}
+	return result
+}
+
+func (b *bitSize) Unmarshal(size interface{}, dimension string, flags uint8) error {
+	for _, dim := range dimensions {
+		if dim.flags&flags == flags && dim.spelling == dimension {
+			// Overflow check
+			switch size := size.(type) {
+			case float64:
+				if float64(^uint64(0)) < size*float64(dim.size) {
+					return errors.New("could not parse bitSize - too big value")
+				}
+				*b = bitSize(size * float64(dim.size))
+				return nil
+			case uint64:
+				if size != 0 && ^uint64(0)/size < uint64(dim.size) {
+					return errors.New("could not parse bitSize - too big value")
+				}
+				*b = bitSize(size) * dim.size
+				return nil
+			default:
+				return errors.New("invalid size for unmarshall, should be float64 or uint64")
+			}
+
+		}
+	}
+	possibleDimensions, _ := json.Marshal(getPossibleDimensions(flags))
+	return fmt.Errorf(`could not parse bitSize - invalid suffix "%s", possible values are %s`,
+		dimension, possibleDimensions)
+}
+
+func (d *BitRate) UnmarshalText(text []byte) error {
+	dimension, size, err := splitDimension(text)
+	if err != nil {
+		return err
+	}
+	//TODO: Do we really want this kind of behaviour?
```

Comment:
No. Raise an error.

Reply:
discussed to leave this for consistency with ByteSize behaviour